crates/holochain/src/conductor/handle.rs

```diff
@@ -276,27 +283,11 @@ impl<DS: DnaStore + 'static> ConductorHandleT for ConductorHandleImpl<DS> {
         lock.add_app_interface_via_handle(port, self.clone()).await
     }
 
-    async fn install_dna(&self, dna: DnaFile) -> ConductorResult<()> {
-        let is_full_wasm_dna = dna
-            .dna_def()
-            .zomes
-            .iter()
-            .all(|(_, zome_def)| matches!(zome_def, ZomeDef::Wasm(_)));
-
-        let mut lock = self.conductor.write().await;
-
-        // Only install wasm if the DNA is composed purely of WasmZomes (no InlineZomes)
-        if is_full_wasm_dna {
-            let entry_defs = lock.put_wasm(dna.clone()).await?;
-            lock.dna_store_mut().add_entry_defs(entry_defs);
-        }
-
-        lock.dna_store_mut().add_dna(dna);
-
-        Ok(())
+    async fn register_dna(&self, dna: DnaFile) -> ConductorResult<()> {
+        self.conductor.write().await.register_dna(dna).await
```

Comment:
Maybe...doesn't seem so bad though, to hold the write lock long enough to do the LMDB write. We can't get around needing the write lock anyway. Should all be quick.

Reply:
I agree if we do [this](https://github.com/holochain/holochain/pull/651#discussion_r578089887) but as long as we are writing the whole dnas bytes to the db in this function then it could be very slow.

Reply:
Also lmdb doesn't need a conductor write lock.

Reply:
Another aspect of this is atomicity. Doesn't it make sense for DNA registration to be an atomic operation? And do we have evidence that writing lots of bytes to LMDB will take a long time?